src/common/syncjournaldb.cpp

```diff
@@ -2287,6 +2302,28 @@ SyncJournalDb::~SyncJournalDb()
     close();
 }
 
+const QVector<SyncJournalFileRecord> SyncJournalDb::getFileRecordsWithDirtyPlaceholders() const
+{
+    QMutexLocker locker(&_mutex);
+
+    if (OC_ENSURE(isOpen())) {
+        const auto query = _queryManager.get(PreparedSqlQueryManager::GetFileReocrdsWithDiryPlaceholdersQuery, getFileRecordQueryC + QByteArrayLiteral("WHERE hasDirtyPlaceholder=TRUE"), const_cast<SyncJournalDb *>(this)->_db);
```

Comment:
is there an index on hasDirtyPlaceholder? This could be a rather big query if the db is big. 

Maybe it is more efficient to have a detail table that _only_ lists the few dirty ones, instead of always checking every entry in metadata?

Reply:
Hmmm https://www.sqlite.org/partialindex.html ?
Adding yet another table and thus adding extra functions to access the table sounds less ideal

Reply:
```
 EXPLAIN QUERY PLAN SELECT path, inode, modtime, type, md5, fileid, remotePerm, filesize,ignoredChildrenRemote, contentchecksumtype.name || ':' || contentChecksum, hasDirtyPlaceholder FROM metadata LEFT JOIN checksumtype as contentchecksumtype ON metadata.contentChecksumTypeId == contentchecksumtype.id WHERE hasDirtyPlaceholder=TRUE;

```

id | parent | notused | detail
-- | -- | -- | --
4 | 0 | 0 | SCAN TABLE metadata USING INDEX hasDirtyPlaceholderIndex
7 | 0 | 0 | SEARCH TABLE checksumtype AS contentchecksumtype USING INTEGER PRIMARY KEY (rowid=?)